### PR TITLE
doc: shields: Add section on using Source Code

### PR DESCRIPTION
**Pull request by ubieda**
### Description

This PR is an [outcome of the discussion](https://github.com/zephyrproject-rtos/zephyr/pull/75310#issuecomment-2258658034) of #75310 in which additional documentation is required to guide users on how to properly utilize source code on shields.

Depends on #75310

### CODE REVIEW

File: doc/hardware/porting/shields.rst [hunk 1]
```diff
@@ -48,6 +48,23 @@ to provide a device nodelabel is the form <device>_<shield>, for instance:
...
+different Zephyr components.
+
+.. note::
+
+   Source Code on Shields shall not be used for purposes other than the
+   one described above. Generic functionalities that could be reused among
+   shields (and/or targets) shall not be captured here.
+
+To effectively incorporate Source code: add a :file:`CMakeLists.txt` file, as
+well as the corresponding source files (referenced on CMake similar to other
```
Comment (kartben) [inline doc/hardware/porting/shields.rst:66]:
```suggestion
well as the corresponding source files (referenced in CMake similar to other
```

File: doc/hardware/porting/shields.rst [hunk 2]
```diff
@@ -48,6 +48,23 @@ to provide a device nodelabel is the form <device>_<shield>, for instance:
                 ...
         };
 
+Adding Source Code
+******************
+
+It is possible to add Source code to Shields, as a way to meet configuration
```
Comment (kartben) [inline doc/hardware/porting/shields.rst:54]: No need for the capital S's here?

File: doc/hardware/porting/shields.rst [hunk 3]
```diff
@@ -48,6 +48,23 @@ to provide a device nodelabel is the form <device>_<shield>, for instance:
...
+requirements that are specific to the shield (e.g: Initialization routines,
+timing constraints, etc); in order to enable it for proper operation with the
+different Zephyr components.
+
+.. note::
+
+   Source Code on Shields shall not be used for purposes other than the
```
Comment (kartben) [inline doc/hardware/porting/shields.rst:61]: Ditto


```suggestion
   Source code in shields shall not be used for purposes other than the
```

File: doc/hardware/porting/shields.rst [hunk 4]
```diff
@@ -48,6 +48,23 @@ to provide a device nodelabel is the form <device>_<shield>, for instance:
...
+   one described above. Generic functionalities that could be reused among
+   shields (and/or targets) shall not be captured here.
+
+To effectively incorporate Source code: add a :file:`CMakeLists.txt` file, as
```
Comment (kartben) [inline doc/hardware/porting/shields.rst:65]:
```suggestion
To effectively incorporate source code: add a :file:`CMakeLists.txt` file, as
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by ubieda** [comment]
Thanks for the feedback @kartben. I've iterated on this PR addressing the change-req's!

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/hardware/porting/shields.rst [hunk 5]
```diff
@@ -48,6 +48,23 @@ to provide a device nodelabel is the form <device>_<shield>, for instance:
...
+
+.. note::
+
+   Source code in shields shall not be used for purposes other than the
+   one described above. Generic functionalities that could be reused among
+   shields (and/or targets) shall not be captured here.
+
+To effectively incorporate source code: add a :file:`CMakeLists.txt` file, as
+well as the corresponding source files (referenced in CMake similar to other
+areas of the SDK, e.g: Boards).
```
Comment (kartben) [inline doc/hardware/porting/shields.rst:67]: just realized the use of the "SDK" term here might not be ideal as the Zephyr SDK is something else - wdyt?

 ```suggestion
well as the corresponding source files (referenced in CMake similar to other
areas of Zephyr, e.g. boards).
```

**Comment by erwango** [comment]
Not blocking but I though the consensus was to document this for boards and shields (ie most of the specification done in boards and in shields would be something like "nothing new on shield vs what is specified for boards for that matter").

File: doc/hardware/porting/shields.rst (hunk 5 above)
Comment (ubieda) [reply doc/hardware/porting/shields.rst]: This makes sense. Applied!

**Review by jfischer-no — APPROVED** [review]
Thanks

**Review by kartben — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**